packages/crashlytics/ios/RNFBCrashlytics/RNFBCrashlyticsInitProvider.m

```diff
@@ -47,6 +47,13 @@ + (BOOL)isCrashlyticsCollectionEnabled {
     enabled = [RNFBMeta getBooleanValue:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED defaultValue:YES];
     DLog(@"isCrashlyticsCollectionEnabled via RNFBMeta: %d", enabled);
   }
+  
+#ifdef DEBUG
+  enabled = NO;
+  if ([[RNFBJSON shared] contains:KEY_CRASHLYTICS_DEBUG_ENABLED]) {
+    enabled = [[RNFBJSON shared] getBooleanValue:KEY_CRASHLYTICS_DEBUG_ENABLED defaultValue:NO];
```

Comment:
🤔 This will override the auto_collection_enabled tests above and is a different logical test than the Java one (which appears correct

At this point enabled might already be no because auto collection is off, and we don't want to turn it on if auto collection is off even if debug collection is on

I think the test needs to be more specific similar to the java test

```suggestion
  if ([[RNFBJSON shared] contains:KEY_CRASHLYTICS_DEBUG_ENABLED] &&
     ![[RNFBJSON shared] getBooleanValue:KEY_CRASHLYTICS_DEBUG_ENABLED defaultValue:NO]) {
    enabled = NO;  
```

I haven't tested this locally - I'm attempting to produce Objective-C code in a web text area so I may not have gotten the syntax / typing correct for the booleans etc but I think it expresses the idea?

Reply:
@mikehardy Your solution will not work as it will enable crashlytics by default as user will not specify KEY_CRASHLYTICS_DEBUG_ENABLED by default. In this case crashlytics will automatically be enabled.

Reply:
I don't understand, I used `&&` and the first part of the clause checks if the key exists. I believe that addresses your concern?

The larger point is the logic needs to be the same Android and your Objective-C solution so far is not. It will potentially overwrite a "wait before collecting / not automatic" false setting with "forget that, we're in debug let's enable it". The settings should have the same precedence behavior on both platforms and on Android if it is set to be not automatic, debug will not enable it. Objective-C should mirror that.

Reply:
Okay - I took a deeper look into this, and the check for contains key is actually unnecessary in this part of the logic because we aren't trying to have different configuration areas take precedence like the auto init settings above, and the default value you provide will be returned if the JSON does not contain the key

So that simplifies things a little bit

Then to match the android implementation I propose *not* setting the value with any defaults in this specific debug-only block, we just set it to false if it is false, that way if it was true or false before from the auto init settings it will start out staying with that value but switch to false unless specifically turned on in debug

so crashes will not be handled in debug by default and auto init settings won't be overridden

I pushed that change to this branch, and after seeing CI go and reviewing it again - if it all still looks good I will merge